syntax = "proto2";

package coolProtocol;


//message of type HostCommand contains an optional command field of type Command xd
message HostCommand {
  enum Command {
    COMMAND_CONNECT = 0;
    COMMAND_DISCONNECT = 1;
    COMMAND_GET_DEVICE_INFO = 2;
  }
  optional Command command= 1;
  
}

message DeviceInfo {
  optional string name = 1;
  optional string os_ver = 2;
  optional string part_number = 3;
  repeated string desc = 4;
}


message Ping {
}

message Pong {
}

// primary message. Only this one is sent over the wire
message MessageWrapper {
  oneof message {
    Ping ping = 1;
    Pong pong = 2;
    HostCommand request = 3;
    DeviceInfo device_data = 4;
  }
}